scripts/check_frontend_coverage.py

```diff
@@ -0,0 +1,378 @@
+    'require-is-float.directive.ts',
+    'review-test-backend-api.service.ts',
+    'review-test-domain.constants.ajs.ts',
+    'review-test-domain.constants.ts',
+    'review-test-page.constants.ajs.ts',
+    'review-test-page.constants.ts',
+    'richTextComponentsRequires.ts',
+    'rich_text_components_definitions.ts',
+    'search-explorations-backend-api.service.ts',
+    'services.constants.ajs.ts',
+    'services.constants.ts',
+    'skill-domain.constants.ajs.ts',
+    'skill-domain.constants.ts',
+    'skill-editor-page.constants.ajs.ts',
+    'skill-editor-page.constants.ts',
+    'skill-mastery-backend-api.service.ts',
+    'skills-mastery-list.constants.ajs.ts',
+    'skills-mastery-list.constants.ts',
+    'state-content.service.ts',
+    'state-editor.constants.ajs.ts',
+    'state-editor.constants.ts',
+    'state-interaction-id.service.ts',
+    'state-recorded-voiceovers.service.ts',
+    'state-rules-stats.service.ts',
+    'state-solicit-answer-details.service.ts',
+    'state-solution.service.ts',
+    'state-top-answers-stats-backend-api.service.ts',
+    'state-written-translations.service.ts',
+    'statistics-domain.constants.ajs.ts',
+    'statistics-domain.constants.ts',
+    'story-domain.constants.ajs.ts',
+    'story-domain.constants.ts',
+    'story-editor-page.constants.ajs.ts',
+    'story-editor-page.constants.ts',
+    'story-viewer-domain.constants.ajs.ts',
+    'story-viewer-domain.constants.ts',
+    'subtopic-viewer-backend-api.service.ts',
+    'subtopic-viewer-domain.constants.ajs.ts',
+    'subtopic-viewer-domain.constants.ts',
+    'summarize-nonnegative-number.filter.ts',
+    'topic-domain.constants.ajs.ts',
+    'topic-domain.constants.ts',
+    'topic-editor-page.constants.ajs.ts',
+    'topic-editor-page.constants.ts',
+    'topic-landing-page.constants.ajs.ts',
+    'topic-landing-page.constants.ts',
+    'topic-viewer-domain.constants.ajs.ts',
+    'topic-viewer-domain.constants.ts',
+    'topics-and-skills-dashboard-domain.constants.ajs.ts',
+    'topics-and-skills-dashboard-domain.constants.ts',
+    'topics-and-skills-dashboard-page.constants.ajs.ts',
+    'topics-and-skills-dashboard-page.constants.ts',
+    'translation-status.service.ts',
+    'translation-tab-active-content-id.service.ts',
+    'translation-tab-active-mode.service.ts',
+    'truncate-at-first-ellipsis.filter.ts',
+    'truncate-at-first-line.filter.ts',
+    'uiLeafletRequires.ts',
+    'underscores-to-camel-case.filter.ts',
+    'undo-redo.service.ts',
+    'user-exploration-permissions.service.ts',
+    'valueGeneratorsRequires.ts',
+    'wrap-text-with-ellipsis.filter.ts',
+]
+
+class LcovStanzaRelevantLines:
+
+    def __init__(self, relevant_lines):
+        """Inicializate the object which provides relevant data of a lcov
+        stanza in order to calculate any decrease in frontend test coverage.
+
+        Args:
+            relevant_lines: list(str). Contains the relevant lines:
```

Comment:
Just pass the whole stanza. You don't need to do a pre-filter; encapsulate all the filtering to get the relevant information within `__init__`.

Reply:
I think this can be a problem. I realized that when `relevant_lines` is filtered in the last iteration of `for item in lcov_items_list`, `relevant_lines` is an empty array. So if I pass the whole stanza, I'm going to create an empty object which is useless. That's why I check `if len(relevan_lines) > 0`, then it'll create objects that really have important information.

Reply:
Before filtering, what does the last item look like? You should probably exclude it.

If it is empty you can get rid of it by calling .strip() on the lcov string before breaking it into chunks.

Reply:
Done.